types/precison_timestamp_types.go

```diff
@@ -0,0 +1,167 @@
+package types
+
+import (
+	"fmt"
+	"github.com/cockroachdb/errors"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type TimePrecision int
```

Comment:
the conversion functions take and return `int32` so why are you using an `int` as the type instead of just defining this as an `int32` ?

Reply:
fixed it, kindly recheck